src/plugins/intel_cpu/src/emitters/x64/jit_eltwise_emitters.cpp

```diff
@@ -428,37 +524,108 @@ void jit_floor_mod_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, con
     Vmm vmm_dst = Vmm(out_vec_idxs[0]);
     Vmm vmm_aux0 = Vmm(aux_vec_idxs[0]);
 
-    if (isa == x64::sse41) {
-        if (vmm_dst.getIdx() != vmm_src0.getIdx())
-            h->uni_vmovups(vmm_dst, vmm_src0);
-        h->uni_vmovups(vmm_aux0, vmm_src0);
-        h->uni_vdivps(vmm_aux0, vmm_aux0, vmm_src1);
-        h->uni_vroundps(vmm_aux0, vmm_aux0, 1); // rounding down
-        h->uni_vmulps(vmm_aux0, vmm_aux0, vmm_src1);
-        h->uni_vsubps(vmm_dst, vmm_dst, vmm_aux0);
-    } else {
-        if (vmm_dst.getIdx() != vmm_src0.getIdx())
-            h->uni_vmovups(vmm_dst, vmm_src0);
-        h->uni_vdivps(vmm_aux0, vmm_src0, vmm_src1);
-        h->uni_vroundps(vmm_aux0, vmm_aux0, 1); // rounding down
-        h->uni_vmulps(vmm_aux0, vmm_aux0, vmm_src1);
-        h->uni_vsubps(vmm_dst, vmm_dst, vmm_aux0);
+    switch (exec_prc_) {
+        case Precision::FP32: {
+            if (isa == x64::sse41) {
+                if (vmm_dst.getIdx() != vmm_src0.getIdx()) {
+                    h->uni_vmovups(vmm_dst, vmm_src0);
+                }
+                h->uni_vdivps(vmm_aux0, vmm_src0, vmm_src1);
+                h->uni_vroundps(vmm_aux0, vmm_aux0, 1); // rounding down
+                h->uni_vmulps(vmm_aux0, vmm_aux0, vmm_src1);
+                h->uni_vsubps(vmm_dst, vmm_dst, vmm_aux0);
+            } else {
+                if (vmm_dst.getIdx() != vmm_src0.getIdx()) {
+                    h->uni_vdivps(vmm_dst, vmm_src0, vmm_src1);
+                    h->uni_vroundps(vmm_dst, vmm_dst, 1); // rounding down
+                    h->vfnmadd132ps(vmm_dst, vmm_src0, vmm_src1);
+                } else {
+                    h->uni_vdivps(vmm_aux0, vmm_src0, vmm_src1);
+                    h->uni_vroundps(vmm_aux0, vmm_aux0, 1); // rounding down
+                    h->vfnmadd231ps(vmm_dst, vmm_aux0, vmm_src1);
+                }
+            }
+        } break;
+        case Precision::I64: {
+            Vmm vmm_aux1 = Vmm(aux_vec_idxs[1]);
```

Comment:
In case of I64 precision and `isa == x64::avx512_core` we need 2 aux registers. Should we account for this in `aux_vecs_count`?